meson.build

```diff
@@ -0,0 +1,48 @@
+# This file is adapted from https://github.com/scipy/scipy/blob/main/meson.build
+project(
+    'pandas',
+    'c', 'cpp', 'cython',
+    version: run_command(['python', 'generate_version.py', '--print'], check: true).stdout().strip(),
+    license: 'BSD-3',
+    meson_version: '>=1.0.1',
+    default_options: [
+        # TODO: investigate, does meson try to compile against debug Python
```

Comment:
Still an issue or outdated comment? I think this is pretty important

Reply:
I assume this was related to a Windows issue, I just got to the bottom of that one. The problem here is that the default buildtype is `debug` when you invoke `meson setup` directly, which does start looking for a Python debug build on Windows. If you use it through `meson-python`, the build type is release, so things are fine. You probably want to add this line to avoid the problem in case someone is trying to use `meson setup` directly:
```
  'buildtype=debugoptimized',
```

Reply:
`buildtype=release` as below is also fine. Meson now has a warning too in case something goes wrong. you can simply delete this comment now.

Reply:
I'll remove in a followup, just to be safe. Any idea why this is a Windows-specific problem?
> 'buildtype=debugoptimized',
Doesn't optimization mess up the debug symbols a little?

Reply:
It is Windows specific because there is MSVC specific code that causes Python.h to force linking to the python_d library instead of the python library, if and only if you compile your own extension with debug synbols.